src/acl.js

```diff
@@ -0,0 +1,84 @@
+export {Allowed} from './attribute/allowed';
+
+export class Acl {
+  permissions = {};
+
+  /**
+   * Set the permissions object with new permissions
+   *
+   * @param {object} permissions
+   *
+   * @chainable
+   */
+  setPermissions(permissions) {
+    this.permissions = {};
+
+    for (let resource in permissions) {
+      if (!permissions.hasOwnProperty(resource)) {
+        continue;
+      }
+
+      this.permit(resource, permissions[resource]);
+    }
+
+    return this;
+  }
+
+  /**
+   * @param {*} resource used to reference the permissions
+   * @param {*} rules used to set the value of the resource
+   *
+   * @chainable
+   */
+  permit(resource, rules) {
+    if (!this.permissions[resource]) {
+      this.permissions[resource] = {};
+    }
+
+    switch (true) {
```

Comment:
The conditionals weren't readable, this is a clear pattern. "If this case is true". This was purely a readability thing.